### PR TITLE
Remove X11 as build dependency, remove numlock detection, rework capslock detection

### PR DESCRIPTION
**Pull request by voidanix**
Makes caps lock actually be recognized under wayland, needs testing on Windows and macOS

### CODE REVIEW

**Comment by TheAssassin** [comment]
Quite a bunch of changes. I hope X11 still works, though?

**Comment by voidanix** [comment]
> Quite a bunch of changes. I hope X11 still works, though?

Works on GNOME X11, should work for other DEs/WMs too

File: src/engine/console.cpp [hunk 1]
```diff
@@ -940,55 +937,17 @@ void writecompletions(stream *f)
...
-    if(d)
-    {
-        uint n = 0;
-        XkbGetIndicatorState(d, XkbUseCoreKbd, &n);
-        XCloseDisplay(d);
-        return (n&0x01)!=0;
-    }
-    #endif
-    return false;
+    if((SDL_GetModState() & KMOD_CAPS) == KMOD_CAPS)
```
Comment (TheAssassin) [inline src/engine/console.cpp:943]: You could also just check whether the result is non-zero. But this works just fine as well.

Why not a bit easier, though?

```cpp
    return (SDL_GetModState() & KMOD_CAPS) == KMOD_CAPS;
```

The return value is `bool`, not `int` anyway...

**Review by TheAssassin — COMMENTED** [review]
Looks good to me mostly, just have one suggestion where you can save 3 LOC.

File: src/engine/console.cpp [hunk 2]
```diff
@@ -940,55 +937,14 @@ void writecompletions(stream *f)
...
-    {
-        uint n = 0;
-        XkbGetIndicatorState(d, XkbUseCoreKbd, &n);
-        XCloseDisplay(d);
-        return (n&0x01)!=0;
-    }
-    #endif
-    return false;
+    // returns 1 if capslock is on
+    return (SDL_GetModState() & KMOD_CAPS) == KMOD_CAPS;
```
Comment (TheAssassin) [inline src/engine/console.cpp:944]: Well, it's actually `true` that is returned. But I guess people get the idea.

**Review by TheAssassin — APPROVED** [review]

**Comment by TheAssassin** [comment]
@voidanix please fix the linking issue on Windows, then we can merge this PR.

**Pull request merged**